### PR TITLE
feat: updated message format for proving updates

### PR DESCRIPTION
**Pull request by dprats**
# Context

## P0: Main intent of this PR

There is a new type in the orchestrator called `ClientProgramProofRequest`. This is replacing the use of `ProverRequest` type. 

### Where it was used

There are four areas where it was used in `prover.rs`

1. registration <-- this PR gets rid of this
2. initial_progress  <-- this PR replaces `ProverRequest` with `ClientProgramProofRequest`
3. progress <-- this PR replaces `ProverRequest` with `ClientProgramProofRequest`
5. response  <-- this PR gets rid of this

## P1: Websockets code

It also cleans up the CLI a bit by moving the websocket code out of `prover.rs` to its own code in `websocket.rs` module and adds a few basic tests for completeness

## P1: `build.rs`

* small change to make makes protobuf compilation in `cargo build` a bit more robust.

### CODE REVIEW

File: clients/cli/src/prover.rs [hunk 1]
```diff
@@ -356,17 +275,22 @@ async fn main() -> Result<(), Box<dyn std::error::Error>>{
...
-                })),
-            };
+  
+            
+
             let progress_duration = progress_time.elapsed();
-            let cycles_proven = steps_proven * 4;
             let proof_cycles_hertz = k as f64 * 1000.0 / progress_duration.as_millis() as f64;
+            
+            let progress = ClientProgramProofRequest {
```
Comment (collinjackson) [inline clients/cli/src/prover.rs:284]: Can we throttle updates to once every 3 minutes? Can do in a separate PR

**Review by collinjackson — APPROVED** [review]

File: clients/cli/src/prover.rs (hunk 1 above)
Comment (dprats) [reply clients/cli/src/prover.rs:284]: Yes let’s do separate PR.
